### ISSUE TITLE
Plugin doesn't support 5.5

### ISSUE THREAD

**Issue by Etamitlucm** [DeviantdVeloper/ButtplugUE#3]
the following modules are missing or built with a different engine version

**Comment by DeviantdVeloper** [comment]
Hi, thanks for getting in touch. Could I bother you for some more details, specifically:

- Exact Engine Version (5.5.X? and is it from the Epic Launcher or built from source)
- Do you have Visual Studio installed (or Xcode if you are on OSX)? You will need to recompile the plugin as I do not provide built binaries currently. I might be able to upload them if needed though.
- Assuming you pressed "Yes" in the message dialog, what output do you get, and also what (if any) log output?

The more info you provide the more likely I can solve the issue 👍

**Comment by Etamitlucm** [comment]
i tried rebuilding it and this is what i get


G:\Games\UE_5.5\Engine\Build\BatchFiles>RunUAT.bat BuildPlugin -plugin="D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\ButtplugUE.uplugin" -package="D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\RebuildedP"
Running AutomationTool...
Using bundled DotNet SDK version: 8.0.300
Starting AutomationTool...
Parsing command line: BuildPlugin -plugin=D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\ButtplugUE.uplugin -package=D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\RebuildedP
Initializing script modules...
Total script module initialization time: 0.32 s.
Using D:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe
Executing commands...
Copying 35 file(s) using max 64 thread(s)
Reading plugin from D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\RebuildedP\HostProject\Plugins\ButtplugUE\ButtplugUE.uplugin...
Building plugin for host platforms: Win64
Running: G:\Games\UE_5.5\Engine\Binaries\ThirdParty\DotNet\8.0.300\win-x64\dotnet.exe "G:\Games\UE_5.5\Engine\Binaries\DotNET\UnrealBuildTool\UnrealBuildTool.dll" UnrealEditor Win64 Development -Project="D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\RebuildedP\HostProject\HostProject.uproject" -plugin="D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\RebuildedP\HostProject\Plugins\ButtplugUE\ButtplugUE.uplugin" -noubtmakefiles -manifest="D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\RebuildedP\HostProject\Saved\Manifest-UnrealEditor-Win64-Development.xml" -nohotreload -log="C:\Users\mrcek\AppData\Roaming\Unreal Engine\AutomationTool\Logs\G+Games+UE_5.5\UBA-UnrealEditor-Win64-Development.txt"
Log file: C:\Users\mrcek\AppData\Roaming\Unreal Engine\AutomationTool\Logs\G+Games+UE_5.5\UBA-UnrealEditor-Win64-Development.txt
Plugin 'ButtplugUE' depends on plugin 'StructUtils' which was deprecated in 5.5 and will soon be removed. Please update your dependencies.
Parsing headers for UnrealEditor
  Running Internal UnrealHeaderTool D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\RebuildedP\HostProject\HostProject.uproject D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\RebuildedP\HostProject\Intermediate\Build\Win64\UnrealEditor\Development\UnrealEditor.uhtmanifest -WarningsAsErrors -installed
D:\Users\mrcek\Downloads\ButtplugUE-main\ButtplugUE-main\RebuildedP\HostProject\Plugins\ButtplugUE\Source\ButtplugUE\Public\BPTypes.h(432): Error: An explicit Category specifier is required for any property exposed to the editor or Blueprints in an Engine module.
Total of 0 written
Total execution time: 8.55 seconds
Took 8.69s to run dotnet.exe, ExitCode=6
UnrealBuildTool failed. See log for more details. (C:\Users\mrcek\AppData\Roaming\Unreal Engine\AutomationTool\Logs\G+Games+UE_5.5\UBA-UnrealEditor-Win64-Development.txt)
AutomationTool executed for 0h 0m 9s
AutomationTool exiting with ExitCode=6 (6)
BUILD FAILED

**Comment by DeviantdVeloper** [comment]
Aha, I see the issue, should be a simple fix. I will make the changes and let you know when it is pushed to the `development` branch where you can grab it and see if it works.

**Comment by DeviantdVeloper** [comment]
Okay, if you go ahead and download from the `development` branch it should be fixed, please let me know how it goes ^_^

**Comment by Etamitlucm** [comment]
maybe im doing something wrong but i cant seem to build it from source 
![image](https://github.com/user-attachments/assets/2374eb21-ddf8-450b-8c78-7f89dad2b2a6)

**Comment by DeviantdVeloper** [comment]
Yeah that is a weird one, I will have to get in there and check what is up, sorry. There might be something else half-finished in `development`, so I will tackle that shortly. Might be a bit slower, given the time of year, but I *am* working on it.

**Comment by DeviantdVeloper** [comment]
Okay so it looks like there were some includes missing/wrong for specifically building the plugin itself: up to now I have only used as-is in the dev project for both development and building the project itself. Another issue I had is that Unreal *really* wanted to package the plugin for iOS, which of course I cannot do on my windows machine.

I have attached below a packaged version of the plugin itself (for Windows specifically), with the problematic headers resolved. Let me know how it goes.

[ButtplugUE.zip](https://github.com/user-attachments/files/18288414/ButtplugUE.zip)

**Comment by Etamitlucm** [comment]
It works!!! thank you for your quick response and the time you dedicated to
help me with my issues and also the debugging menu is switching the devices
around when picked if you have two devices atleast for me when devise 0 is
clicked it shows the settings for device 1 and the same thing goes when
picking device 1

**Comment by DeviantdVeloper** [comment]
Glad I got something sorted, but okay that other issue is a new one, I will look into it ASAP as it sounds pretty breaking, I will let you know how it goes shortly.

**Comment by DeviantdVeloper** [comment]
So I tried to test this here but in Editor, Dev and Shipping builds I could not reproduce this, however I have a very limited set of devices I can control, namely two Xbox controllers with identical names. The commands were reaching the correct devices however, so I suspect that it is either a bug with the UI itself, or something weird happening with the device names themselves.

Could you tell me exactly what you are doing and what you are experiencing, including what the device names are in Intiface itself versus in the demo UI?

**Comment by Etamitlucm** [comment]
Xbox controller and the handy

**Comment by DeviantdVeloper** [comment]
Hrm well leave it with me and I will look into it.

At the very least, commands are being sent properly to the correct devices, so you can work and develop as needed, and the cosmetic issue of incorrect names can be resolved and patched in the meantime.

**Comment by Etamitlucm** [comment]
I have recorded a video showing the issue.
https://github.com/user-attachments/assets/e1c65c2e-eb8b-4aa9-a49f-70a452b3dc8a

**Comment by DeviantdVeloper** [comment]
Hmm yeah okay I see the issue and it is very odd. Seems like all the information is there, just that the buttons and info screens or opposite of one another.

Could you indulge me and tell me what order they are listed in on Intiface, and also what happens when you try to move the sliders in UE to make them move?

**Comment by Etamitlucm** [comment]
so after seeing your comment i figured out the solution to my problem
![image (35)](https://github.com/user-attachments/assets/5078e9fb-b54f-4815-9b55-5268ca1e5a0f)
![image (36)](https://github.com/user-attachments/assets/94a83a64-86cf-4d05-ba47-d986bea8f70e)
i just reversed it XD

https://github.com/user-attachments/assets/513be431-1de8-4293-8761-e0cf9885cf10

**Comment by Etamitlucm** [comment]
everything's working perfectly now. XD I feel kinda dumb though, I'm still new to coding and struggling with the patterns.

**Comment by DeviantdVeloper** [comment]
Hey that looks great, well done.

No problem at all. That solution feels a little band-aidy, so I will continue looking into it, but I am glad it is working for you now ^_^

Thanks again for raising this, it really does help me make the plugin better.
